apps/token/src/routes/staking/associate/wallet-associate.tsx

```diff
@@ -0,0 +1,112 @@
+import './wallet-associate.scss';
+
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { TokenInput } from '../../../components/token-input';
+import { ADDRESSES } from '../../../config';
+import {
+  AppStateActionType,
+  useAppState,
+} from '../../../contexts/app-state/app-state-context';
+import { useContracts } from '../../../contexts/contracts/contracts-context';
+import { TxState } from '../../../hooks/transaction-reducer';
+import { useTransaction } from '../../../hooks/use-transaction';
+import { BigNumber } from '../../../lib/bignumber';
+import { AssociateInfo } from './associate-info';
+import type { VegaKeyExtended } from '@vegaprotocol/wallet';
+
+export const WalletAssociate = ({
+  perform,
+  vegaKey,
+  amount,
+  setAmount,
+  address,
+}: {
+  perform: () => void;
+  amount: string;
+  setAmount: React.Dispatch<string>;
+  vegaKey: VegaKeyExtended;
+  address: string;
+}) => {
+  const { t } = useTranslation();
+  const {
+    appDispatch,
+    appState: { walletBalance, allowance, walletAssociatedBalance },
+  } = useAppState();
+
+  const { token } = useContracts();
+
+  const {
+    state: approveState,
+    perform: approve,
+    dispatch: approveDispatch,
+  } = useTransaction(() => token.approve(ADDRESSES.stakingBridge));
+
+  // Once they have approved deposits then we need to refresh their allowance
+  React.useEffect(() => {
+    const run = async () => {
+      if (approveState.txState === TxState.Complete) {
+        const allowance = await token.allowance(
+          address,
+          ADDRESSES.stakingBridge
+        );
+        appDispatch({
+          type: AppStateActionType.SET_ALLOWANCE,
+          allowance,
+        });
+      }
+    };
+    run();
+  }, [address, appDispatch, approveState.txState, token]);
+
+  let pageContent = null;
+
+  if (
+    walletBalance.isEqualTo('0') &&
+    // eslint-disable-next-line
+    new BigNumber(walletAssociatedBalance!).isEqualTo('0')
+  ) {
+    pageContent = (
+      <div className="wallet-associate__error">
+        {t(
+          'You have no VEGA tokens in your connected wallet. You will need to buy some VEGA tokens from an exchange in order to stake using this method.'
+        )}
+      </div>
+    );
+  } else if (
+    walletBalance.isEqualTo('0') &&
+    // eslint-disable-next-line
```

Comment:
?

Reply:
Defaulted walletAssociatedBalance to zero